### PR TITLE
Add note about validating an array of hashes

### PR DESCRIPTION
**Pull request by georgemillo**

### CODE REVIEW

File: source/gems/dry-validation/nested-data.html.md [hunk 1]
```diff
@@ -89,3 +89,30 @@ puts errors.inspect
...
+      required(:age).filled(:int?, gteq?: 18)
+    end
+  end
+end
+
+errors = schema.call(
+  people: [ { name: 'Alice', age: 19 }, { name: 'Bob', age: 17 } ],
+).messages
+
+errors = schema.call(phone_numbers: ['123456789', 123456789]).messages
```
Comment (solnic) [inline source/gems/dry-validation/nested-data.html.md]: Could you change this to use `errors` instead of `messages`?

Comment (georgemillo) [inline source/gems/dry-validation/nested-data.html.md]: I can do, but on closer inspection I've realised: is there any functional difference between these two schemas?

```
Dry::Validation.Schema do
  required(:people).each do
    required(:name).filled(:str?)
    required(:age).filled(:int?, gteq?: 18)
  end
end

Dry::Validation.Schema do
  required(:people).each do
    schema do
      required(:name).filled(:str?)
      required(:age).filled(:int?, gteq?: 18)
    end
  end
end
```

Are these both approaches that should be documented?

Comment (solnic) [inline source/gems/dry-validation/nested-data.html.md]: @georgemillo yes, the former won't be possible in 1.0.0 :)

Comment (georgemillo) [inline source/gems/dry-validation/nested-data.html.md]: Okay, made the requested updates and rebased :)

**Pull request merged**